Update migrations to not use Mix

klappradla opened the pull request:
Mix is not available at compile time. So migrations checking the env via
Mix.env() don't work for our releases to Fly.io.

klappradla commented:
Ok wow, this is a disaster. It's running in test (wrong) and having a non-null constraint violation 😱

andreasknoepfle commented:
@klappradla I would argue, that it was a bad idea in the first place adding a conditional on the environment for this. 

This kind of thing should be in the seeds, rather than in a migration, so you have more control when it is executed. 

TLDR: I would remove the insert entierly

klappradla committed:
Update env-dependent migrations

This adds a helper function to figure out the environment a migration is
run in, regardless of at compile or at runtime.

This feels a bit like a hack and maybe we should adopt a pattern like
those:
- https://blog.appsignal.com/2020/02/25/migrating-production-data-in-elixir.html
- https://dashbit.co/blog/automatic-and-manual-ecto-migrations

klappradla commented:
Saw your comment too late @andreasknoepfle 🙈 Just updated the code.

I guess we would need some pattern like https://dashbit.co/blog/automatic-and-manual-ecto-migrations or https://blog.appsignal.com/2020/02/25/migrating-production-data-in-elixir.html

So you say, add it to the seeds (I did already) and manually run them on staging/production after the deploy? I modified one of the seed calls though - that probably needs change. Should I then add running the seeds to the `Release` module, so that it's always done on every deploy? If they are idempodent, we could do it like this, right?

klappradla commented:
Closing this as it's superseeded by #105

All in all, it's an anti-pattern to insert data via migrations. It makes more sense to do this with specific data-migrations or one-off tasks.
